agreement.md

```diff
@@ -0,0 +1,151 @@
+
+1. Proposals MUST be governed by this Agreement, or a future version of this Agreement as modified within the bounds of the Agreement.
+
+2. Disputes over a proposal relying on this Agreement MUST be resolved by Aragon Court.
+
+3. Proposals MUST NOT disproportionately benefit a majority of ANT holders voting on a proposal over the minority.
+
+4. Proposals MUST NOT attempt to cast a vote using ANT held in the address of any versions of the Agent app installed on the Aragon Network organization, or using ANT held in any Network contract that fulfills similar functionality as the Agent app.
+
+5. Proposals to amend either the current Aragon Court Controller contract or any new Controller contract that may be proposed for Aragon Court MUST NOT amend Aragon Court configurations, funds, and modules governors to be any entity other than an Agent app installed on the Aragon Network organization.
+
+6. If a proposal deploys a new Aragon Court Jurors Registry contract and/or amends Aragon Court to use a new Jurors Registry, then the new Jurors Registry MUST use ANJ as the native juror token.
+
+7. Proposals MUST NOT activate in Aragon Court any ANJ controlled by the Aragon Network.
+
+8. Proposals MUST NOT use the Aragon Network organization’s role as Aragon Court funds governor to transfer ANJ staked and/or activated by Aragon Court jurors to another Ethereum account.
+
+## Section 3. Mutable guidelines
+The guidelines in this section MAY be amended. These guidelines apply to all proposals put forth to the Aragon Network organization.
+
+1. Proposals MUST be consistent with the Aragon Manifesto.
+
+2. Proposals to transfer assets from the Aragon Network organization to the exclusive control of any other entity SHALL NOT be allowed until Ethereum block #15,000,000.
+
+3. Proposals MUST be accompanied by an accurate American English-language Justification for the proposal. Typos in the Justification SHALL be allowed if they do not substantially alter the meaning of the proposal. Each Justification MUST be referred to in the on-chain source code or transaction data of the proposal by the content-addressed location of the Justification on the public IPFS network, and the Justification MUST be accessible via the public IPFS network for as long as the proposal is open for voting and/or in effect. If a Justification becomes temporarily inaccessible via the public IPFS network, and its proposal is challenged under this guideline, then jurors SHOULD block the proposal if they have reason to believe the Justification was inaccessible at the time that the proposal was challenged. Alternatively, the Justification MAY be encoded directly into the on-chain source code or transaction data of the proposal so that the Justification remains human-readable and accessible as long as the Ethereum blockchain itself remains publicly available. The Justification MUST include the following information:
```

Comment:
Is there a specific reason for forcing people to use IPFS instead of simply requiring that the proposal be kept publicly accessible?

Reply:
I think the file will be more likely to stay online if it's hosted on a network that supports multiple seeders, and since Aragon uses IPFS for a number of other purposes already it is a natural fit.

Reply:
Distributed/redundant storage preferred, no doubt. I just started wondering whether something could be said here about lock-in, optionality, flexibility. If there's a technical reason why it has to be IPFS then, of course, it makes sense.

Reply:
@mlphresearch 

IPFS is a content-addressable storage, so the proposal can not be changed without changing its URL. If there are other storage networks that have this property, I think they can be allowed as well.

Reply:
@xuhcc that is another nice property, yes. Also it was pointed out that the Aragon Court Dashboard uses IPFS to fetch the Justification, which is displayed as the first argument for a proposal. So having the requirement that the Justification be uploaded to and available via IPFS makes sense so it can be easily consumed by the Court Dashboard.